### ISSUE TITLE
[RMP] Cross-framework evaluation metrics

### ISSUE THREAD

**Issue by karlhigley** [NVIDIA-Merlin/Merlin#407]
# Problem statement
Merlin models should allow you to train models from different frameworks (TF/PT/lightfm/xgboost/implicit etc) and then compare them in a consistent "apples-to-apples" type way. Right now each framework has its own evaluation metrics, that aren't directly comparable to other frameworks. During the paper experiments for instance, we had to implement custom metric code for implicit to match the output from TF.  Customers also want to be able to compare to their own metrics.

# Goals
- Customers are able to compare models across frameworks.  
- Coverage of metrics: Precision, NDCG, MAP, AUC
- Support for both retrieval and ranking models
- Only used during final evaluation process.  This isn't used during training. (Non goal) 
- Example 

# Constraints
- Scoring should happen for all items, not just negative samples, in order to be able to compare correctly for ranking metric
- Only one implementation of these metrics so that the calculations are consistent.  This implies conversion.
- We want to be able to compare across multiple frameworks.

# Notes 
- Gabriel has a POC of cupy code: https://github.com/rapidsai/recsys/tree/main/benchmark_recsys/code/evaluation 
- Marc built a system like this at Spotify which converted to specific framework metrics

# Tasks
## 22.07 ( Ground work for cross framework evaluation )

- [ ] Create a mechanism for transferring data across frameworks
  - [ ] https://github.com/NVIDIA-Merlin/core/pull/102 

## 22.08 ( Enter the goal here )

- [ ] Establish a standard set of metrics (to be used in conjunction with cross-framework data transfer)
- [ ] Create example for cross framework evaluation metrics

**Comment by karlhigley** [comment]
Duplicate of NVIDIA-Merlin/models#450

**Comment by karlhigley** [comment]
Closing https://github.com/NVIDIA-Merlin/models/issues/450 instead

**Comment by viswa-nvidia** [comment]
Based on discussion with Karl, the tasks are not fully listed down

**Comment by viswa-nvidia** [comment]
@marcromeyn please review definition and add tasks for 22.11 and 22.12 - starting point specifically

**Comment by viswa-nvidia** [comment]
@marcromeyn , please add information about the PoC in the ticket
